internal/pkg/term/progress/component.go

```diff
@@ -15,6 +15,15 @@ const (
 	nestedComponentPadding = 2 // Leading space characters for rendering a nested component.
 )
 
+// noopComponent satisfies the Renderer interface but does not write anything.
+type noopComponent struct{}
```

Comment:
Is it upcoming deployment component stub struct?

Reply:
It's mostly a component to remove a [mode](https://a-nickels-worth.dev/posts/modesharm/) from the code. For example, without this struct the code would look like:
```go
if deploymentRenderer != nil {
   ...
} else {
   return nl, nil
}
```
whereas if we provide a dummy struct that just satisfies the interface but doesn't have any behavior:
```go
var r Renderer = &noopComponent{}
if deploymentRenderer != nil {
   r = deploymentRenderer
}
...
```
We got rid of the else-statement :P